### ISSUE TITLE
A lot of "out of bounds" errors

### ISSUE THREAD

**Issue by lifeofguenter** [GigaOM/go-sphinx#39]
I get a lot of the following errors:

```
go-sphinx wpdb error: sphinx_query(): sphinxql error: offset out of bounds (offset=3680, max_matches=1000) for query: SELECT id, post_date_gmt FROM wp_posts_delta,wp_posts WHERE ( post_author = 5 ) AND MATCH( '@post_type post @post_status publish' ) ORDER BY post_date_gmt DESC LIMIT 3680, 1000 OPTION ranker=proximity_bm25, max_matches=1000;
```

Using Wordpress 4.1.1 / Nginx 1.7.10 / PHP-FPM 5.5.22 / Sphinx 2.2.8

**Comment by misterbisson** [comment]
@lifeofguenter you're right to point this one out. It occurs when people page through results beyond the first 1000 matching results. Balancing this is a matter of performance vs. the utility of being able to see the entire dataset.

If the results are date sorted, it would be possible to add date as a criteria to get older results than the first 1000, but our experience was such that it never rose to the priority that would justify our time.

I'd love to see a pull request if you have some ideas for a performant solution, though.

**Comment by lifeofguenter** [comment]
@misterbisson I haven't checked in all detail where your plugin replaces the default behaviour - a normal pagination (e.g. blog.org/page/1, blog.org/page/..., blog.org/page/1111) should never be affected by such, and instead should revert back to the "normal" mysql query.

I totally agree with you though, the search itself should be in most cases useless to provide more than 1000 results and should rather force the user to "refine" his search.

That is not the point though. The point is, that it this needs to be catched prior. It should not happen that a mysqli_query results in such an error - especially if it can be simply avoided by checking something like $requested_offset > 1000 then do not even bother sphinx.

**Comment by misterbisson** [comment]
> a normal pagination (e.g. blog.org/page/1, blog.org/page/..., blog.org/page/1111) should never be affected by such, and instead should revert back to the "normal" mysql query.

I don't recall if that search is replaced or not. The peculiarities of how it was used at http://search.gigaom.com didn't require that. I agree it probably shouldn't interfere with such searches, though.

> this needs to be catched prior. It should not happen that a mysqli_query results in such an error - especially if it can be simply avoided by checking something like $requested_offset > 1000

The problem is that the limit is a Sphinx server-side config item, not a client-side value that can be tested before the query. The value could be co-configured in the plugin and tested as you suggest, but we didn't know how significance the problem was at first and it was just as likely we would have had to build support for paging past the first 1000 results. So letting it error, and handling the error in a not too miserable way was the best solution to tracking it and allowing us to decide later.
